inc/classes/subscriber/Tools/class-cloudflare-subscriber.php

```diff
@@ -0,0 +1,888 @@
+						/* translators: %1$s = opening link; %2$s = closing link */
+						__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+						// translators: Documentation exists in EN, FR; use localized URL if applicable.
+						'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
+						'</a>'
+					);
+					return new \WP_Error( 'cloudflare_wrong_zone_id', $msg );
+				}
+
+				return true;
+			}
+		} catch ( \Exception $e ) {
+			$msg  = __( 'Incorrect Cloudflare email address or API key.', 'rocket' );
+			$msg .= ' ' . sprintf(
+				/* translators: %1$s = opening link; %2$s = closing link */
+				__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+				// translators: Documentation exists in EN, FR; use localized URL if applicable.
+				'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
+				'</a>'
+			);
+			return new \WP_Error( 'cloudflare_invalid_auth', $msg );
+		}
+	}
+
+	/**
+	 * Get a Cloudflare\Api instance & the zone_id corresponding to the domain
+	 *
+	 * @since 2.8.21 Get the zone ID from the options
+	 * @since 2.8.18 Add try/catch to prevent fatal error Uncaugh Exception
+	 * @since 2.8.16 Update to Cloudflare API v4
+	 * @since 2.5
+	 *
+	 * @return Object Cloudflare instance & zone_id if credentials are correct, WP_Error otherwise
+	 */
+	public static function get_instance() {
+		$cf_email             = get_rocket_option( 'cloudflare_email', null );
+		$cf_api_key           = ( defined( 'WP_ROCKET_CF_API_KEY' ) ) ? WP_ROCKET_CF_API_KEY : get_rocket_option( 'cloudflare_api_key', null );
+		$cf_zone_id           = get_rocket_option( 'cloudflare_zone_id', null );
+		$is_api_keys_valid_cf = get_transient( 'rocket_cloudflare_is_api_keys_valid' );
+
+		if ( false === $is_api_keys_valid_cf ) {
+			$is_api_keys_valid_cf = self::is_api_keys_valid( $cf_email, $cf_api_key, $cf_zone_id );
+			set_transient( 'rocket_cloudflare_is_api_keys_valid', $is_api_keys_valid_cf, 2 * WEEK_IN_SECONDS );
+		}
+
+		if ( is_wp_error( $is_api_keys_valid_cf ) ) {
+			return $is_api_keys_valid_cf;
+		}
+
+		$cf_api_instance = new \Cloudflare\Api( $cf_email, $cf_api_key );
+		$cf_api_instance->setCurlOption( CURLOPT_USERAGENT, self::CF_USER_AGENT );
+
+		$cf_instance = (object) [
+			'auth'    => $cf_api_instance,
+			'zone_id' => $cf_zone_id,
+		];
+
+		return $cf_instance;
+	}
+
+	/**
+	 * Get all the current Cloudflare settings for a given domain.
+	 *
+	 * @since 2.8.16 Update to Cloudflare API v4
+	 * @since 2.5
+	 *
+	 * @return mixed bool|Array Array of Cloudflare settings, false if any error connection to Cloudflare
+	 */
+	public static function get_settings() {
+		$cf_instance = self::get_instance();
```

Comment:
Then, and for all the others:

```suggestion
		$cf_instance = self::get_cloudflare_instance();
```